### ISSUE TITLE
Flow analysis bug with ternary operator

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#277]
```
The nullness checker incorrectly reports a possible null dereference in the 
following snippet.

I tested with the 1.7.0 release.

===

import checkers.nullness.quals.*;

abstract class Ternary {
  void f(@Nullable Object o) {
    g(42, o != null ? o.hashCode() : 0);
  }

  abstract void g(Object x, Object xs);
}

===
```

Original issue reported on code.google.com by `cus...@google.com` on 21 Nov 2013 at 2:11
